framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -1295,6 +1300,25 @@ public void postAsMemberOf(
             addAnnotationFromFieldInvariant(type, owner, (VariableElement) element);
         }
         addComputedTypeAnnotations(element, type);
+        if (viewpointAdaptor != null && viewpointAdaptor.shouldBeAdapted(type, element)) {
+            AnnotatedTypeMirror decltype = this.getAnnotatedType(element);
+            AnnotatedTypeMirror combinedType =
+                    this.viewpointAdaptor.combineTypeWithType(owner, decltype, this);
+            // TODO Remove this ugly code
+            // Update: we don't have a method to do this job. I think we need to add this logic.
+            type.replaceAnnotation(viewpointAdaptor.getAnnotationMirror(combinedType, this));
+            if (type.getKind() == TypeKind.DECLARED
+                    && combinedType.getKind() == TypeKind.DECLARED) {
+                AnnotatedDeclaredType adtType = (AnnotatedDeclaredType) type;
+                AnnotatedDeclaredType adtCombinedType = (AnnotatedDeclaredType) combinedType;
+                adtType.setTypeArguments(adtCombinedType.getTypeArguments());
+            } else if (type.getKind() == TypeKind.ARRAY
+                    && combinedType.getKind() == TypeKind.ARRAY) {
+                AnnotatedArrayType aatType = (AnnotatedArrayType) type;
+                AnnotatedArrayType aatCombinedType = (AnnotatedArrayType) combinedType;
+                aatType.setComponentType(aatCombinedType.getComponentType());
```

Comment:
I'm not quite sure about the context. So just want to ping here --- is this code need to recursively set component type for multi-demension array type? (Like the bug I fixed a few weeks ago in GTIS).